src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -171,6 +171,9 @@ public TimeSpan FirstBroadcastAttemptTimeout
             }
         }
 
+        public event EventHandler<HttpRequestMessage> OnBeforeRequest;
+        public event EventHandler<HttpResponseMessage> OnAfterRequests;
```

Comment:
Not a good idea, we need to send the request as well to the after callback.
Otherwise, not way to correlate

Reply:
https://github.com/ravendb/ravendb/pull/11013#discussion_r471467897